pil_analyzer/src/condenser.rs

```diff
@@ -160,6 +160,19 @@ impl<T: FieldElement> Condenser<T> {
             }
             Expression::UnaryOperation(op, inner) => {
                 let inner = self.condense_expression(inner);
+                if *op == UnaryOperator::Next {
+                    if let AlgebraicExpression::Reference(reference) = inner {
+                        assert!(!reference.next, "Double application of \"'\"");
```

Comment:
We could just allow this here and change `next: bool` to `shift: int32` or something. But this would need substantial changes in the witness generator as well.